### PR TITLE
[NO GBP] Stops External Shields From Generating On Solid Walls 

### PR DESCRIPTION
**Pull request by moocowswag**
## About The Pull Request

#75705 (I swear to god i tested this) caused external shields to generate over solid walls, this adds the check necessary to prevent that and I tested to make sure it was fixed WITHOUT CAUSING UNINTENDED BEHAVIORS
## Why It's Good For The Game

Okay so ya this bug is not really impactful and it does actually play into the role of external shields which is to not block internal pathways and instead protect the already existing physical structure of the station/provide separate structures with further autonomy. 

The problem mainly is its not intended and its completely different behavior than what was proposed in the original pr that added shield gens

That and it just looks really bad and ugly to have forcefields appear over solid walls.

## Changelog
:cl:

fix: external shields no longer generate over walls

/:cl:

### CODE REVIEW

File: code/game/machinery/modular_shield.dm [hunk 1]
```diff
@@ -178,7 +178,7 @@
 
 		if(exterior_only)
 			for(var/turf/target_tile as anything in list_of_turfs)
-				if(!(isfloorturf(target_tile)) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
+				if(!(isfloorturf(target_tile)) && isopenturf(target_tile) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
```
Comment (MrMelbert) [inline code/game/machinery/modular_shield.dm:181]: Can you change this to a few early continues instead? It's getting pretty cluttered at this point
```dm
if(isfloorturf(target_tile))
    continue
if(!isopenturf(target_tile))
    continue
if(locate(/obj/structure/emergency_shield/modular) in target_tile)
    continue
...
```

File: code/game/machinery/modular_shield.dm [hunk 2]
```diff
@@ -199,7 +199,7 @@
 	LAZYADD(inside_shield, circle_range_turfs(src, radius - 1))//in the future we might want to apply an effect to the turfs inside the shield
 	if(exterior_only)
 		for(var/turf/target_tile as anything in circle_range_turfs(src, radius))
-			if(!(isfloorturf(target_tile)) && !(target_tile in inside_shield) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
+			if(!(isfloorturf(target_tile)) && isopenturf(target_tile) && !(target_tile in inside_shield) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
```
Comment (MrMelbert) [inline code/game/machinery/modular_shield.dm:202]: Ditto

**Comment by MrMelbert** [comment]
> You might aswell type the for loop to turf/open if you're going to be adding a type check for every loop

This is also so true

**Comment by moocowswag** [comment]
Im close to just trying to see if i can blacklist tiles that have something in dense in them at all.

**Comment by moocowswag** [comment]
> You might aswell type the for loop to turf/open if you're going to be adding a type check for every loop

Ya thats the smartest way to do this pr thanks.

**Comment by moocowswag** [comment]
uhhhh its spawning over walls when i do for(var/turf/open/target_tile as anything in list_of_turfs)

**Comment by moocowswag** [comment]
tested

doesnt form on solid walls on either mode ? check
forms on nonfloor turfs on both modes? check
only forms on floor turfs on internal mode? check

**Pull request merged**